runner/operator/tempo_nextflow_operator/tempo_nextflow_operator.py

```diff
@@ -0,0 +1,410 @@
+            return data
+        else:
+            new_str = unicodedata.normalize("NFKD", data)
+            return new_str
+        return ""
+
+    def create_tracker_file(self):
+        """
+        Creates the string for tracker
+
+        String is tab-delimited; special consideration taken so that the first two columns
+        is the Tumor/Normal pairing (if the row Sample is a tumor) or
+        Normal/"N/A" (if row sample is a normal)
+
+        The rest of the columns follow the metadata field names in the order set in lists key_order and extra_keys
+
+        key_order has specifically formatted header values, as defined by the PMs, so they needed to be separate;
+        extra_keys values are the metadata field names in the database, used as headers
+        """
+        tracker = ""
+        key_order = ["investigatorSampleId", "sampleName", "sampleClass"]
+        key_order += ["baitSet", settings.REQUEST_ID_METADATA_KEY]
+        extra_keys = [
+            "tumorOrNormal",
+            "species",
+            settings.RECIPE_METADATA_KEY,
+            settings.SAMPLE_CLASS_METADATA_KEY,
+            settings.SAMPLE_ID_METADATA_KEY,
+            settings.PATIENT_ID_METADATA_KEY,
+        ]
+        extra_keys += [
+            "investigatorName",
+            "investigatorEmail",
+            "piEmail",
+            "labHeadName",
+            "labHeadEmail",
+            "preservation",
+        ]
+        extra_keys += ["dataAnalystName", "dataAnalystEmail", "projectManagerName", "investigatorId", "cmoSampleName"]
+
+        tracker = "CMO_Sample_ID\tCollaborator_ID_(or_DMP_Sample_ID)\tHistorical_Investigator_ID_(for_CCS_use)\tSample_Class_(T/N)\tBait_set_(Agilent/_IDT/WGS)\tIGO_Request_ID_(Project_ID)\t"
+        for key in extra_keys:
+            tracker += key + "\t"
+        tracker = tracker.strip() + "\n"
+
+        seen = set()
+
+        for patient_id in self.patients:
+            patient = self.patients[patient_id]
+            for sample_id in patient.all_samples:  # must do this to account for unpaired normals
+                sample = patient.all_samples[sample_id]
+                meta = sample.dedupe_metadata_values()
+                if sample.cmo_sample_name not in seen:
+                    seen.add(sample.cmo_sample_name)
+                    if not sample.conflict:  # metadata is valid sample
+                        running = list()
+                        running.append(sample.cmo_sample_name)
+                        for key in key_order:
+                            v = meta[key]
+                            s = self._validate_to_str(v)
+                            running.append(s)
+                        for key in extra_keys:
+                            s = self._validate_to_str(meta[key])
+                            running.append(s)
+                        tracker += "\t".join(running) + "\n"
+        return self.write_to_file("sample_tracker.txt", tracker)
```

Comment:
This tracker file was being used by Tempo team to populate the CCS' WES tracker HERA, used by the PMs.

We need to check with CCS to see if this is going to need continued support in the future as I believe HERA is at or approaching end-of-life.